### PR TITLE
Support for showing playback progress in media_player

### PR DESCRIPTION
**Pull request by bjarniivarsson**
This adds support for showing playback progress in the media_player UI.

This uses playback progress information provided in a separate PR to home-assistant: https://github.com/home-assistant/home-assistant/pull/4172

### CODE REVIEW

**Comment by bjarniivarsson** [comment]
Empty job log on travis :(

**Comment by robbiet480** [comment]
Restarted travis

**Comment by robbiet480** [comment]
I don't know why it's continuing to output an empty job log, @balloob will need to take a look.

File: src/cards/ha-media_player-card.html [hunk 1]
```diff
@@ -148,6 +157,13 @@
     </div>
 
     <div class='controls layout horizontal justified'>
+      <paper-progress
+        max=[[playbackDuration]]
+        value=[[playbackPosition]]
+        invisible$='[[!playbackVisible]]'
```
Comment (balloob) [inline src/cards/ha-media_player-card.html]: please use `hidden$` instead.

File: src/cards/ha-media_player-card.html [hunk 2]
```diff
@@ -148,6 +157,13 @@
     </div>
 
     <div class='controls layout horizontal justified'>
+      <paper-progress
+        max=[[playbackDuration]]
```
Comment (balloob) [inline src/cards/ha-media_player-card.html]: Please wrap attributes in quotes.

File: src/cards/ha-media_player-card.html [hunk 3]
```diff
@@ -219,13 +235,75 @@
...
+    playbackPosition: {
+      type: Number,
+    },
+
+    playbackVisible: {
+      type: Boolean,
+      value: false
+    },
+
+    _internal: {
```
Comment (balloob) [inline src/cards/ha-media_player-card.html]: If it is not going to be used inside the templates or as callbacks, no need to add it as attributes.

File: src/cards/ha-media_player-card.html [hunk 4]
```diff
@@ -219,13 +235,75 @@
...
+    }
+    else {
+      this._internal.position = null;
+      this.playbackDuration = null;
+      this._internal.time = null;
+    }
+
+    if (playerObj.isPlaying && this._internal.position != null && this.playbackDuration) {
+      // playback is in progress, update the playback status on an interval timer
+      this._internal.interval = setInterval(this.updateProgress.bind(this), 500);
```
Comment (balloob) [inline src/cards/ha-media_player-card.html]: Why change interval every 0.5 seconds while the progress bar has a 1 second granularity?

**Review by balloob — COMMENTED** [review]
Also some linting errors.

**Comment by bjarniivarsson** [comment]
@balloob the linting errors are in unrelated files. Regarding the 500ms interval comment from you, that was an attempt to make the movement of the progress bar smooth, I changed it to 200 to make it even smoother.

**Comment by bjarniivarsson** [comment]
Accidentally closed :)

**Comment by justweb1** [comment]
@robbiet480 @balloob It doesn't look like these travis errors should be issues. They are simply a single line of code that has been wrapped.

**Pull request merged**